File: src/main/java/com/mhl/mycompanybackend/repository/PermissionsRepository.java
package com.mhl.mycompanybackend.repository;

import com.mhl.mycompanybackend.model.PermissionName;
import com.mhl.mycompanybackend.model.Permissions;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface PermissionsRepository extends JpaRepository<Permissions, Long> {
    Optional<Permissions> findByName(PermissionName name);
}

File: src/main/java/com/mhl/mycompanybackend/repository/PermissionsRepository.kt
package com.mhl.mycompanybackend.repository

import com.mhl.mycompanybackend.model.PermissionName
import com.mhl.mycompanybackend.model.Permissions
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface PermissionsRepository : JpaRepository<Permissions?, Long?> {
    fun findByName(name: PermissionName?): Optional<Permissions?>?
}
